grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -154,25 +230,42 @@ class ProductAvailabilityService {
         String binLocationName = entry?.binLocation?.name ?
                 "'${StringEscapeUtils.escapeSql(entry?.binLocation?.name)}'" : "'DEFAULT'"
 
-        def insertStatement =
+        // NOTE: I had to split it to avoid unnecessary overwriting of quantity allocated and on hold
```

Comment:
Because I had two versions of quantities returned by bin location I wanted to avoid setting 0s in allocated and on hold qty whenever it was not required. Now I moved calculations to InventoryService, this is no longer required.